sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/metadata/MetadataContext.java

```diff
@@ -0,0 +1,119 @@
+ * configured by {@linkplain System#getenv(String) environment variable} or {@linkplain
+ * System#getProperty(String) system property} {@code METADATA_HEADERS} as comma separated list of
+ * header names, e.g. {@code business-process-id,tenant-id}. Header names are treated
+ * case-insensitive.
+ *
+ * <p>Technically, the {@code MetadataContext} is implemented as static {@link ThreadLocal}. When
+ * code is executed in a new {@link Thread}, the context must be transferred. Helper methods are
+ * available for {@linkplain #transferMetadataContext(Runnable) <code>Runnable</code>} and
+ * {@linkplain #transferMetadataContext(Callable) <code>Callable</code>}
+ */
+// TODO not picked up from request headers and Kafka consumers as documented yet.
+// TODO Not handled in platform client and KafkaMessageProducer as documented yet.
+public interface MetadataContext {
+
+  /**
+   * @return the current immutable metadata context, never {@code null}.
+   */
+  static MetadataContext current() {
+    return MetadataContextUtil.current();
+  }
+
+  /**
+   * @return the current metadata context, never {@code null}. Changes in the returned instance will
+   *     not affect the {@link MetadataContext} of the current {@link Thread}.
+   */
+  static DetachedMetadataContext detachedCurrent() {
+    return DetachedMetadataContext.of(MetadataContextUtil.current());
+  }
+
+  /**
+   * Creates a new {@link MetadataContext} for the current {@link Thread}. Any existing {@link
+   * MetadataContext} in the current {@link Thread} will be replaced.
+   *
+   * @param metadataContext the new {@link MetadataContext}, e.g. {@linkplain
+   *     DetachedMetadataContext#toMetadataContext() derived} from a {@link
+   *     DetachedMetadataContext}.
+   */
+  static void createContext(DetachedMetadataContext metadataContext) {
+    MetadataContextHolder.set(metadataContext.toMetadataContext());
+  }
+
+  /**
+   * Transfers the current metadata context to the runnable when executed in a new thread.
+   *
+   * @param runnable The runnable to wrap with the current metadata context.
+   * @return The original runnable wrapped with code to transfer the metadata context when executed
+   *     in a new thread.
+   */
+  static Runnable transferMetadataContext(Runnable runnable) {
+    return MetadataContextUtil.transferMetadataContext(runnable);
+  }
+
+  /**
+   * Transfers the current metadata context to the callable when executed in a new thread.
+   *
+   * @param callable The runnable to wrap with the current metadata context.
+   * @return The original callable wrapped with code to transfer the metadata context when executed
+   *     in a new thread.
+   */
+  static <V> Callable<V> transferMetadataContext(Callable<V> callable) {
+    return MetadataContextUtil.transferMetadataContext(callable);
+  }
+
+  /**
+   * @return all available keys in the metadata context.
+   */
+  Set<String> metadataKeys();
```

Comment:
I'd prefer `keys()` as method name. Does it clash with implementing another interface?

Reply:
I'll try.